fedimint-core/src/db/mod.rs

```diff
@@ -723,6 +723,28 @@ impl<'isolated, T: MaybeSend + Encodable> ModuleDatabaseTransaction<'isolated, T
             })
     }
 
+    #[instrument(level = "debug", skip_all, fields(key = ?key_prefix))]
+    pub async fn find_by_prefix_sorted_descending<KP>(
```

Comment:
Guessing keys are stored in order so this is constant time?

Reply:
Kind of, afaik all our DB backends have some sort of BTree index, so lookups should be `O(log(n))` and ordered traversal probably `O(1)`? @m1sterc001guy